Update livestreaming docs (#147)
fishjam-cloud/documentation: docs/explanation/public-livestreams.mdx

---
type: explanation
title: Private vs Public Livestreams
sidebar_position: 5
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

# Private vs Public Livestreams

:::info

> This explanation only concerns Fishjam rooms that have type `livestream`.

Reviewer: **nitpick (blocking):** or just `the type`
```suggestion
This explanation only applies to Fishjam rooms of type `livestream`.
```

---
type: explanation
title: Private vs Public Livestreams
sidebar_position: 5
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

# Private vs Public Livestreams

:::info
This explanation only concerns Fishjam rooms that have type `livestream`.
:::

> When livestreaming with Fishjam, you have the ability to chose whether your livestream is `private` or `public`.

Reviewer 2: **nitpick:** 
```suggestion
When livestreaming with Fishjam, you have the choice to make your livestream `private` or `public`.
```

By default, created livestreams are `private`, to prevent possible bugs.

<Tabs groupId="language">
  <TabItem value="ts" label="Typescript">
  
    ```ts
    const fishjamUrl = "";
    const managementToken = "";

    // ---cut---
    import { FishjamClient } from '@fishjam-cloud/js-server-sdk';

    const fishjamClient = new FishjamClient({
      fishjamUrl,
      managementToken,
    });

    const privateRoom = await fishjamClient.createRoom({ roomType: 'livestream' }); // [!code highlight]
    const publicRoom = await fishjamClient.createRoom({ roomType: 'livestream', public: true }); // [!code highlight]
    ```

  </TabItem>

  <TabItem value="python" label="Python">

    ```python
    from fishjam import FishjamClient

    fishjam_client = FishjamClient(

>       fishjam_url=fishjam_url,

Reviewer 2: **issue:** update to use fishjam id

Reviewer 2: here an in every other example

      management_token=management_token,
    );

    private_room = await fishjam_client.create_room(room_type="livestream") # [!code highlight]
    public_room = await fishjam_client.create_room(room_type="livestream", public=True) # [!code highlight]
    ```

  </TabItem>
</Tabs>

> The choice you make affects how **viewers** connect to the livestream and has no affect on streamers.

Reviewer 2: **nitpick:** is `has no affect on streamers` purposeful? or should it be `effect`?

Reviewer 2: ```suggestion
The choice you make affects how **viewers** connect to the livestream and has no effect on streamers.
```

    const fishjamClient = new FishjamClient({
      fishjamUrl,
      managementToken,
    });

    const privateRoom = await fishjamClient.createRoom({ roomType: 'livestream' });
    const { token } = await fishjamClient.createLivestreamViewerToken(privateRoom.id); // [!code highlight]
    ```

  </TabItem>

  <TabItem value="python" label="Python">

    ```python
    from fishjam import FishjamClient

    fishjam_client = FishjamClient(
      fishjam_url=fishjam_url,
      management_token=management_token,
    );

    private_room = await fishjam_client.create_room(room_type="livestream")
    token = await fishjam_client.create_livestream_viewer_token(private_room.id) # [!code highlight]
    ```

  </TabItem>
</Tabs>

### Connecting to a private room

Once you've created a viewer token, you can connect to a room using the Fishjam client SDKs (examples below), or alternatively you can use [WHEP](/how-to/features/whip-whep#private-livestreams).

<Tabs groupId="platform">
  <TabItem value="react" label="React">
  
    ```tsx
    const viewerToken = "";

    // ---cut---
    import { useLivestreamViewer } from '@fishjam-cloud/react-client';

>     const { connect } = useLivestreamViewer();
> 
>     await connect({ token: viewerToken });

Reviewer 2: **nitpick:** 
```suggestion
    // stream will be available upon connecting
    const { connect, stream } = useLivestreamViewer();

    await connect({ token: viewerToken });
```
